### ISSUE TITLE
gto which: No such command

### ISSUE THREAD

**Issue by francesco086** [iterative/gto#252]
After upgrading to gto v0.2.2 (from 0.2.1) my production pipeline ended with the error:
```
No such command 'which'. 
```

On my local I got the same error, and a `gto -h` returned the following list of commands
```
╭─ Commands 
│ annotate              Update artifact metadata annotations                                                                                                                                           │
│ assign                Assign stage to specific artifact version                                                                                                                                      │
│ check-ref             Find out the artifact version registered/assigned with ref                                                                                                                     │
│ deprecate             Deprecate artifact, deregister a version, or unassign a stage                                                                                                                  │
│ describe              Display enrichments for an artifact                                                                                                                                            │
│ history               Show a journal of registry operations                                                                                                                                          │
│ register              Create an artifact version to signify an important, published or released iteration                                                                                            │
│ remove                Remove the enrichment for given artifact                                                                                                                                       │
│ show                  Show the registry state                                                                                                                                                        │
│ stages                Print list of stages used in the registry
```

`which` has disappeared...?

**Comment by francesco086** [comment]
@aguschin I think I tracked the source of the problem: https://github.com/iterative/gto/commit/4576fa0e40c3c288133aa9c42a6199aaf4fbebb5#diff-a398e421c291a116d32b2020f4d98324054f61b653b81ab032ba006317de2001L638

What was the logic behind removing these lines?

**Comment by aguschin** [comment]
Yep, they were removed in favor of `gto show`. Now instead of `gto latest nn` one should call `gto show nn@greatest`, and instead of `gto which nn prod` one should call `gto show nn#prod`. It was introduced in https://github.com/iterative/gto/pull/244

I was required to add options to sort assignments differently, and realized that what `which` and `latest` can do is a subset of `show`. Uniting them leaves less commands in place (which is clearer IMO) and makes some command examples shorter (like using `model#dev` instead of `model --stage dev` which could be introduced in some cases).

Considering your feedback, I think I needed to keep the commands and issue "deprecated" warnings to not confuse users and break workflows that are in place already. Will be more careful next time. Thanks for bringing this up!
